File: core/src/main/java/javabot/commands/DropApi.java
package javabot.commands;

import com.antwerkz.sofia.Sofia;
import javabot.Message;
import javabot.dao.ApiDao;
import javabot.javadoc.JavadocApi;
import org.pircbotx.Channel;

import javax.inject.Inject;

public class DropApi extends AdminCommand {
    @Inject
    private ApiDao dao;
    @Param
    String name;

    @Override
    public void execute(final Message event) {
        final JavadocApi api = dao.find(name);
        if (api != null) {
            drop(event, api);
        } else {
            getBot().postMessageToChannel(event, Sofia.unknownApi(name, event.getUser().getNick()));
        }
    }

    private void drop(final Message event, final JavadocApi api) {
        getBot().postMessageToChannel(event, Sofia.adminRemovingOldJavadoc(api.getName()));
        dao.delete(api);
        getBot().postMessageToChannel(event, Sofia.adminDoneRemovingOldJavadoc(api.getName()));
    }
}

File: src/main/kotlin/javabot/commands/DropApi.kt
package javabot.commands

import com.antwerkz.sofia.Sofia
import com.beust.jcommander.Parameter
import javabot.Javabot
import javabot.Message
import javabot.dao.AdminDao
import javabot.dao.ApiDao
import javabot.javadoc.JavadocApi
import javax.inject.Inject

class DropApi @Inject constructor(bot: Javabot, adminDao: AdminDao, var apiDao: ApiDao) : AdminCommand(bot, adminDao) {
    @Parameter(required = true)
    lateinit var apiName: String

    override fun execute(event: Message): List<Message> {
        val responses = arrayListOf<Message>()
        val api = apiDao.find(apiName)
        if (api != null) {
            drop(responses, event, api)
        } else {
            responses.add(Message(event, Sofia.unknownApi(apiName, event.user.nick)))
        }
        return responses
    }

    private fun drop(responses: MutableList<Message>, event: Message, api: JavadocApi) {
        responses.add(Message(event, Sofia.adminRemovingOldJavadoc(api.name)))
        apiDao.delete(api)
        responses.add(Message(event, Sofia.adminDoneRemovingOldJavadoc(api.name)))
    }
}